### ISSUE TITLE
Error posting to Slack - peer not authenticated

### ISSUE THREAD

**Issue by RolandBurrows** [jenkinsci/slack-plugin#837]
### Jenkins and plugins versions report

<details>
  <summary>Environment</summary>

Slack plugin version: 616.v03b_1e98d13dd
Jenkins version: 2.346.2 (LTS)
  - Controller: 1
  - Agents: 5
  - Pipelines type: declarative Jenkinsfiles

</details>

### What Operating System are you using (both controller, and any agents involved in the problem)?

Controller and Agents instance information:
  - OS: Windows Server 2019 Datacenter (fully up to date)
  - Java: openJDK 11.0.2
  - Hosting: AWS EC2

### Reproduction steps

In our self-hosted instance of Jenkins, a Jenkins core upgrade (from 2.346.1) was performed, along with a handful of plugin updates (list not available, apologies). The day afterwards, pipelines began to fail intermittently on the `slackSend()` plugin command, with the following error message appearing in the jenkins.err logs:
    
```
WARNING j.p.slack.StandardSlackService#postToSlack: Error posting to Slack
javax.net.ssl.SSLPeerUnverifiedException: peer not authenticated
```
    
Pipelines with multiple `slackSend()` commands whose errors are caught, alternatingly work as expected. Error / Send / Error / Send.

Rollbacks to the previous stage did not restore functionality.
    
Additionally, in the 'Configure System` screen, 'Slack' panel:
   - the 'Test Connection' button if repeatedly pressed generates alternating `Success` / `Failure` / `Success` / `Failure` / `Success` / `Failure` responses.
   - if 'Custom slack app bot user' is selected and the 'Test Connection' button is repeatedly pressed, it generates the following response sequence:  `Failure` / `Failure: {"ok":false,"error":"invalid_auth"}` / `Failure` / `Failure: {"ok":false,"error":"invalid_auth"}`.
    
We have disabled Slack messages until we can figure out root cause, but it's a core feature we heavily rely upon (and accordingly appreciate all the work that goes into developing / maintaining this plugin).
    
Please advise!
    

### Expected Results

Plugin would not error on sending Slack messages.

### Actual Results

Plugin errors on sending Slack messages.

**Comment by timja** [comment]
Sounds like a networking issue, guessing it's something to do with an SSL inspecting / MiTM proxy you have.

https://stackoverflow.com/a/13146619/4951015

**Comment by RolandBurrows** [comment]
A solution has been found!

We suspect that the following happened:
- Slack (the company) has a load balancer in front of their API.
- Slack made changes to one of their proxy-clusters related to certificates (as a canary?).
- We believe that these changes were incompatible with our installed openJDK version regarding its certificate library.
- This created the round-robin effect we saw where it would alternate between pass / fail, reproducibly.
- Thus, we updated the openJDK version from 11.0.2 to 11.0.15 and it fixed the issue!

Thanks for taking the time to think about the issue we ran into.
